hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/SmartClientInvocationService.java

```diff
@@ -91,6 +91,7 @@ public void invokeOnPartitionOwner(ClientInvocation invocation, int partitionId)
             throw new IOException("Partition does not have an owner. partitionId: " + partitionId);
         }
         if (!isMember(owner)) {
+            partitionService.firstArrangement();
```

Comment:
this may also be redundant due to line 89

Reply:
This is necessary in certain scenarios. It could be the case that partition table is triggered on master node. Client gets the partition table. Then master dies before sending partition table to other nodes. A second node here does not aware of ` firstArrangement` . In that case, we need to trigger `firstArragement` again.